be/src/storage/txn_manager.cpp

```diff
@@ -245,20 +252,35 @@ Status TxnManager::persist_tablet_related_txns(const std::vector<TabletSharedPtr
     SCOPED_RAW_TIMER(&duration_ns);
 
     std::unordered_set<std::string> persisted;
+    std::vector<TabletSharedPtr> to_flush_tablet;
     for (auto& tablet : tablets) {
         if (tablet == nullptr) {
             continue;
         }
         auto path = tablet->data_dir()->path();
         // skip persisted meta.
         if (persisted.find(path) != persisted.end()) continue;
+        to_flush_tablet.push_back(tablet);
+        persisted.insert(path);
+    }
 
-        auto st = tablet->data_dir()->get_meta()->flush();
+    auto token = _thread_pool_flush->new_token(ThreadPool::ExecutionMode::CONCURRENT);
+    std::vector<std::pair<Status, int64_t>> pair_vec(to_flush_tablet.size());
+    int i = 0;
+    for (auto& tablet : to_flush_tablet) {
+        auto dir = tablet->data_dir();
+        token->submit_func([&pair_vec, dir, i]() { pair_vec[i].first = std::move(dir->get_meta()->flush()); });
```

Comment:
how about the queue is full and fail to submit_func?

Reply:
The threadpool is created with an unlimited queue by default which is ok in this situation i think.